app/src/components/common/big_number_input/index.tsx

```diff
@@ -41,92 +41,59 @@ const Input = styled.input`
   }
 `
 
-export class BigNumberInput extends React.Component<Props, State> {
-  public static defaultProps = {
-    placeholder: '0.00',
-  }
-
-  public readonly state = {
-    currentValueStr: this.props.value
-      ? ethers.utils.formatUnits(this.props.value, this.props.decimals)
-      : '',
-    currentDecimalsStr: this.props.decimals,
-  }
-
-  private textInput: any
-
-  public static getDerivedStateFromProps = (props: Props, state: State) => {
-    const { decimals, value } = props
-    const { currentValueStr, currentDecimalsStr } = state
-
-    if (!value) {
-      return {
-        currentValueStr: '',
-      }
-    } else if (
-      value &&
-      !ethers.utils.parseUnits(currentValueStr || '0', currentDecimalsStr).eq(value)
-    ) {
-      return {
-        currentValueStr: ethers.utils.formatUnits(value, decimals),
-      }
-    } else {
-      return null
+export const BigNumberInput = (props: Props) => {
```

Comment:
This doesn't have the exact same behavior as before. For example, if the value is `0.0`, and you press backspace, nothing happens. That's why in the previous implementation there was a state variable `currentValueStr`: you need to keep track of the string in the input, and transform it in a BigNumber for these scenarios to work.